src/main/java/ch/jalu/configme/resource/yaml/SnakeYamlNodeBuilderImpl.java

```diff
@@ -0,0 +1,159 @@
+    }
+
+    protected @NotNull Node createNumberNode(@NotNull Number value) {
+        Tag tag = (value instanceof Double || value instanceof Float || value instanceof BigDecimal)
+            ? Tag.FLOAT
+            : Tag.INT;
+        return new ScalarNode(tag, value.toString(), null, null, DumperOptions.ScalarStyle.PLAIN);
+    }
+
+    protected @NotNull Node createBooleanNode(boolean value) {
+        return new ScalarNode(Tag.BOOL, String.valueOf(value), null, null, DumperOptions.ScalarStyle.PLAIN);
+    }
+
+    protected @NotNull Node createSequenceNode(@NotNull Stream<?> entries, @NotNull String path,
+                                               @NotNull ConfigurationData configurationData) {
+        AtomicInteger counter = new AtomicInteger();
+        String pathPrefix = path.isEmpty() ? "" : path.concat(".");
+
+        List<Node> values = entries
+            .map(entry -> {
+                String entryPath = pathPrefix.concat(Integer.toString(counter.getAndIncrement()));
+                return toYamlNode(entry, entryPath, configurationData, 0);
+            })
+            .collect(Collectors.toList());
+
+        return new SequenceNode(Tag.SEQ, values, DumperOptions.FlowStyle.BLOCK);
+    }
+
+    protected @NotNull Node createMapNode(@NotNull Map<String, ?> value, String path,
+                                          @NotNull ConfigurationData configurationData) {
+        String pathPrefix = path.isEmpty() ? "" : path.concat(".");
+        List<NodeTuple> nodeEntries = new ArrayList<>(value.size());
+
+        for (Map.Entry<String, ?> entry : value.entrySet()) {
+            Node keyNode = createKeyNode(entry.getKey());
+            Node valueNode = toYamlNode(entry.getValue(), pathPrefix.concat(entry.getKey()), configurationData, 0);
+            transferComments(valueNode, keyNode);
+
+            nodeEntries.add(new NodeTuple(keyNode, valueNode));
+        }
+
+        return new MappingNode(Tag.MAP, nodeEntries, DumperOptions.FlowStyle.BLOCK);
+    }
+
+    /**
+     * Creates comments based on all possible sources (number of empty lines, configuration data,
+     * {@link ValueWithComments}) and returns them as SnakeYAML comment lines.
+     *
+     * @param value the export value
+     * @param path the path the value is located at
+     * @param configurationData the configuration data instance
+     * @param numberOfNewLines number of new lines to add to the beginning of the comments
+     * @return comment lines representing all defined comments
+     */
+    protected @NotNull List<CommentLine> collectComments(@NotNull Object value, @NotNull String path,
+                                                         @NotNull ConfigurationData configurationData,
+                                                         int numberOfNewLines) {
+        Stream<String> emptyLineStream = IntStream.range(0, numberOfNewLines)
+            .mapToObj(i -> "\n");
```

Comment:
Should we add a util for this? I see this for the second time and hate how this statement looks. Shame there isn't something like `Stream.repeat("\n", numberOfNewLines)`